src/main/java/com/google/devtools/build/lib/runtime/BlazeWorkspace.java

```diff
@@ -395,6 +406,23 @@ public ActionCache getOrLoadPersistentActionCache(Reporter reporter) throws IOEx
                 runtime.getClock(),
                 reporter);
       }
+
+      // Here we clear any server-lifetime entries in the action cache that a previous instance of
+      // the server might have created.
+      if (clearActionCacheServerLifetimeEntries) {
+        // We only exercise this code at most once during the lifetime of the server.
+        clearActionCacheServerLifetimeEntries = false;
+        try (AutoProfiler p = profiledAndLogged("pruning server-lifetime entries from action cache", ProfilerTask.INFO)
```

Comment:
How long does this take on a larger repo? It's a cost paid even by someone who doesn't use the new feature, so we need to be careful.